internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,194 @@
+package daemonset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	daemonSetName    = "partner-repo"
+	namespace        = "default"
+	containerName    = "container-00"
+	imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
+	timeout          = 5 * time.Minute
+	nodeExporter     = "node-exporter"
+)
+
+//nolint:funlen
+func CreateDaemonSetsTemplate(dsName, namespace, containerName, imageWithVersion string) *v1.DaemonSet {
```

Comment:
You could configure the security context here by passing a bool parameter to this function.
`func CreateDaemonSetsTemplate(dsName, namespace, containerName, imageWithVersion string, runAsPrivileged bool)` and when runAsPrivileged is true, then use make `** Privileged: true**.

Reply:
I was looking at our existing debug daemonset YAML [here](https://github.com/test-network-function/cnf-certification-test-partner/blob/main/test-partner/debugpartner.yaml#L30) and the privileged pod setting is always true.

Reply:
I changed the 'runAsPrivileged' to be 'True' always.